addons/knowledge/static/src/js/widgets/knowledge_dialogs.js

```diff
@@ -0,0 +1,186 @@
+                            records.results.push(result);
+                            continue;
+                        }
+                        const children = result.children.filter(child => {
+                            const text = child.text.toLowerCase();
+                            const term = data.term.toLowerCase();
+                            return text.indexOf(term) >= 0;
+                        });
+                        if (children.length > 0) {
+                            records.results.push({...result, children});
+                        }
+                    }
+                    if (data.results.length > 0) {
+                        records.results.push({
+                            text: _t('Articles'),
+                            children: data.results.map(record => {
+                                return {
+                                    id: record.id,
+                                    icon: record.icon,
+                                    text: record.name,
+                                }
+                            })
+                        });
+                    }
+                    return records;
+                },
+            },
+            /**
+             * @param {Object} data
+             * @param {JQuery} container
+             * @param {Function} escapeMarkup
+             */
+            formatSelection: (data, container, escapeMarkup) => {
+                const markup = [];
+                if (data.id === 'private') {
+                    const src = escapeMarkup(this.getLoggedUserPicture());
+                    markup.push(`<img src="${src}" class="rounded-circle mr-1"/>`);
+                } else if (typeof data.icon !== 'undefined') {
+                    markup.push(`<i class="fa fa-w ${escapeMarkup(data.icon)} mr-1"/>`);
+                }
+                markup.push(escapeMarkup(data.text));
+                return markup.join('');
+            },
+            /**
+             * @param {Object} result
+             * @param {JQuery} container
+             * @param {Object} query
+             * @param {Function} escapeMarkup
+             */
+            formatResult: (result, container, query, escapeMarkup) => {
+                const { text } = result;
+                const markup = [];
+                Select2.util.markMatch(text, query.term, markup, escapeMarkup);
+                if (result.id === 'private') {
+                    const src = escapeMarkup(this.getLoggedUserPicture());
+                    markup.unshift(`<img src="${src}" class="rounded-circle mr-1"/>`);
+                } else if (typeof result.icon !== 'undefined') {
+                    markup.unshift(`<i class="fa fa-w ${escapeMarkup(result.icon)} mr-1"/>`);
+                }
+                return markup.join('');
+            },
+        });
+    },
+
+    /**
+     * @override
+     */
+    save: function () {
```

Comment:
This does not seem to "save" anything?
It just calls the callback function received in the init?